### PR TITLE
Updated csv fresh classroom import script to include classroom space …

### PR DESCRIPTION
**Pull request by jerrywu64**
…types

### CODE REVIEW

File: esp/useful_scripts/csv_fresh_classroom_import.py [hunk 1]
```diff
@@ -68,12 +84,94 @@ def get_available_furnishings():
...
         break
     else:
         RESOURCE_MATCHING = {}
 
+SPACE_TYPE_MATCHING = {}
+def match_space_type(space_type):
+    if space_type == "":
+        return ""
+    # http://stackoverflow.com/questions/3673434/find-subsequences-of-strings-within-strings
+    pattern = ".*".join(space_type)
```
Comment (benjaminjkraft) [inline esp/useful_scripts/csv_fresh_classroom_import.py]: I'm confused why we are finding a subsequence here?

File: esp/useful_scripts/csv_fresh_classroom_import.py [hunk 2]
```diff
@@ -68,12 +84,94 @@ def get_available_furnishings():
...
+        for space_type in space_types:
+            match_attempt = match_space_type(space_type)
+            if match_attempt is None or force_manual:
+                possible_types = get_possible_space_Types()
+                if force_manual:
+                    print "Manual mode forced due to attempt failure."
+                print ("Unable to automatically match space type "
+                "{} to possible space types.").format(space_type)
+                print "Available space types:"
+                for space in possible_types:
```
Comment (benjaminjkraft) [inline esp/useful_scripts/csv_fresh_classroom_import.py]: May want to `for i, space in enumerate(possible_types): print i, space` to make typing the index easy.  (Similarly a couple other places.)

File: esp/useful_scripts/csv_fresh_classroom_import.py [hunk 3]
```diff
@@ -68,12 +84,94 @@ def get_available_furnishings():
...
+                    print space
+                match_attempt = raw_input(
+                    "Input space type index (or 'None'): ")
+                if match_attempt == "None":
+                    match_attempt = ""
+                else:
+                    match_attempt = POSSIBLE_SPACE_TYPES[
```
Comment (benjaminjkraft) [inline esp/useful_scripts/csv_fresh_classroom_import.py]: probably want a try/except here so it doesn't crash if they typo

File: esp/useful_scripts/csv_fresh_classroom_import.py [hunk 4]
```diff
@@ -68,12 +84,94 @@ def get_available_furnishings():
...
+    matches = []
+    for possible_type in POSSIBLE_SPACE_TYPES:
+        if re.search(pattern, possible_type):
+            matches.append(possible_type)
+    if len(matches) == 1:
+        return matches[0]
+    else:
+        return None
+
+def get_possible_space_Types():
```
Comment (benjaminjkraft) [inline esp/useful_scripts/csv_fresh_classroom_import.py]: nit: lowercase `T`

File: esp/useful_scripts/csv_fresh_classroom_import.py (hunk 2 above)
Comment (jerrywu64) [reply esp/useful_scripts/csv_fresh_classroom_import.py]: It's already taken care of as part of get_possible_space_types() but I'll rewrite to make it clearer.

**Comment by jerrywu64** [comment]
Wait, why is it failing?

**Comment by betaveros** [comment]
Is this #2388 again?

**Comment by jerrywu64** [comment]
recreating with rebase onto main with #2439.

**Pull request closed without merging**